ros/src/twist_controller/twist_controller.py

```diff
@@ -1,14 +1,40 @@
-
+import rospy
+from pid import PID
+from yaw_controller import YawController
 GAS_DENSITY = 2.858
 ONE_MPH = 0.44704
 
 
 class Controller(object):
-    def __init__(self, *args, **kwargs):
-        # TODO: Implement
-        pass
-
-    def control(self, *args, **kwargs):
-        # TODO: Change the arg, kwarg list to suit your needs
-        # Return throttle, brake, steer
-        return 1., 0., 0.
+    def __init__(self, vehicle_mass, decel_limit, accel_limit, wheel_radius,
+                 wheel_base, steer_ratio, max_lat_accel, max_steer_angle):
+        self.velocity_pid = PID(0.2, 0.0, 3, mn=decel_limit, mx=accel_limit)
```

Comment:
why remove integral from PID?

Reply:
Sorry, I removed it for testing to avoid issues with integral component growing indefinitely. I can put it back if you want but seems to work fine like this.

Reply:
the sim car probably doesn't have misaligned wheels but carla might. lets return some small value here

Reply:
A value of 0.004 results in a much worse behavior in the simulator. The car exceeds the target speed (45 mph vs 40) and for some reason it has even weaker steering (???)

I'll keep it to 0 for now and we can tune it later.

Reply:
see comment about 'ki' above. I think I understand why you are correct leaving it 0